From cc7224752030ebaa426ec7926f98cb5396fc166d Mon Sep 17 00:00:00 2001
From: David Ng <dave@codeaurora.org>
Date: Wed, 11 Apr 2018 10:43:57 -0700
Subject: [PATCH 26/26] Explicitly allow system_server to (m)map data files

Linux kernel 4.14+ SELinux starts explicit map
permission check for file mmap operations.  Add this
permission to system_server for data file access,
which is used in scenario such as "adb install" of
APK's.

test: no longer see SELinux map denial on "adb install"
Bug: 113238007
Change-Id: Id6016dd0b3f15dfdb0f02509ea812dee61ac78ed
Merged-In: Id6016dd0b3f15dfdb0f02509ea812dee61ac78ed
(cherry picked from commit 383471c267b6792f1625f8f771d8e0c0b9090300)
---
 private/system_server.te | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/private/system_server.te b/private/system_server.te
index a1ea12f9..bbdd7ea5 100644
--- a/private/system_server.te
+++ b/private/system_server.te
@@ -455,7 +455,14 @@ allow system_server system_app_data_file:file create_file_perms;
 
 # Receive and use open app data files passed over binder IPC.
 # Types extracted from seapp_contexts type= fields.
-allow system_server { system_app_data_file bluetooth_data_file nfc_data_file radio_data_file shell_data_file app_data_file }:file { getattr read write append };
+allow system_server {
+  system_app_data_file
+  bluetooth_data_file
+  nfc_data_file
+  radio_data_file
+  shell_data_file
+  app_data_file
+}:file { getattr read write append map };
 
 # Access to /data/media for measuring disk usage.
 allow system_server media_rw_data_file:dir { search getattr open read };
-- 
2.17.1

